### DISCUSSION TITLE
Eglo / AwoX led driver

### DISCUSSION THREAD

**Discussion by taugusti** [Koenkk/zigbee2mqtt#18366] [Devices]
I have finally found a way to make the Eglo / AwoX zigbee lights work properly.
Their customer support advised me to use another app - not the one listed in the documentation online and with the product :-(
"AwoX HomeControl" is the correct app. It included a more recent firmware version for the lamps - and it works without blocking the newwork and generating the infamous NWK_TABLE_FULL errors.
The new fw version is 2.5.0_999.
But... they report themselves with a new zigbee id, so a new converter is required.

I looked at the "old" eglo/awox driver - and it just does an extend of light_onoff_brightness_colortemp:
"extend: extend.light_onoff_brightness_colortemp({colorTempRange: [153, 370]})"
So - I did the same in a custom converter and it works fine.
on/off, brightness, color_temp and color_temp_startup, power_on_behavior works fine.
The effects also does "something" - not sure if it is as expected. I have never used that feature before so I do not know what to expect...

But - the question is if that is the right extend to use? When I go to the "exposes" tab in zigbee2mqtt, there is a block called color_xy - but it is not a color LED lamp...
<img width="444" alt="image" src="https://github.com/Koenkk/zigbee2mqtt/assets/15052792/577a85fc-01d8-4463-98b8-9cec208f98e6">

Do I misunderstand the color_xy?
Can someone help me ensure the converter is correct?
What information do you need?

Here the list of zigbee endpoints as reported by zigbee2mqtt:

Endpoint 1
 Output clusters
  genOnOff
 Input clusters
  genBasic
  genIdentify
  genGroups
  genScenes
  genOnOff
  genLevelCtrl
  lightingColorCtrl
  touchlink
  64599
  genTime
Endpoint 3
 Output clusters
  65360
  65361
 Input clusters
  65360
  65361
  genGroups
Endpoint 242
 Output clusters
  greenPower

Thanks!

**Comment by Koenkk** [comment]
That's great news!

Can you provide the data/database.db entry of this device and as which model it should be recognized?

**Reply by taugusti** [reply]
Hi @Koenkk 
The devices I have are two different Eglo HERRORA-Z models - 60x60cm and 30x120cm:
30x120: https://www.eglo.com/uk/ceiling-light-herrora-z-99549.html
60x60: https://www.eglo.com/uk/ceiling-light-herrora-z-99548.html
there is also a 45x45 version that I assume is the same: https://www.eglo.com/uk/ceiling-light-herrora-z-99639.html

Inside they have the same AwoX LED driver.

btw. I updated using the app and bluetooth, so I am not sure we can get that fw into Zigbee2Mqtt directly?

And the line from the database:
`{"id":104,"type":"Router","ieeeAddr":"0xa4c1380584a54b84","nwkAddr":59624,"manufId":4417,"manufName":"AwoX","powerSource":"Mains (single phase)","modelId":"EGLO_ZM_TW_CLP","epList":[1,3,242],"endpoints":{"1":{"profId":260,"epId":1,"devId":268,"inClusterList":[0,3,4,5,6,8,768,4096,64599,10],"outClusterList":[6],"clusters":{"genBasic":{"attributes":{"modelId":"EGLO_ZM_TW_CLP","manufacturerName":"AwoX","powerSource":1,"zclVersion":3,"appVersion":2,"stackVersion":1,"hwVersion":60,"swBuildId":"2.5.0_999"}},"genOnOff":{"attributes":{"onOff":0}},"lightingColorCtrl":{"attributes":{"colorCapabilities":16,"colorTempPhysicalMin":153,"colorTempPhysicalMax":370}}},"binds":[{"cluster":6,"type":"endpoint","deviceIeeeAddress":"0xa4c1380584a54b84","endpointID":1},{"cluster":6,"type":"endpoint","deviceIeeeAddress":"0x00124b0022988f3a","endpointID":1}],"configuredReportings":[{"cluster":6,"attrId":0,"minRepIntval":0,"maxRepIntval":3600,"repChange":1}],"meta":{}},"3":{"profId":4751,"epId":3,"devId":268,"inClusterList":[65360,65361,4],"outClusterList":[65360,65361],"clusters":{},"binds":[],"configuredReportings":[],"meta":{}},"242":{"profId":41440,"epId":242,"devId":97,"inClusterList":[],"outClusterList":[33],"clusters":{},"binds":[],"configuredReportings":[],"meta":{}}},"appVersion":2,"stackVersion":1,"hwVersion":60,"swBuildId":"2.5.0_999","zclVersion":3,"interviewCompleted":true,"meta":{"configured":2},"lastSeen":1689916819446,"defaultSendRequestWhen":"immediate"}
`

**Reply by Koenkk** [reply]
Given the `modelId`, I would expect the device to be detected as a [99847](https://github.com/Koenkk/zigbee-herdsman-converters/blob/2bea3af1cec4020f727ff6131f51b054ed1ba8ff/src/devices/eglo.ts#L30) which only supports color temp (at least in z2m 1.32.1), isn't this the case?

**Reply by taugusti** [reply]
ha ha - apparently someone added it while I was messing around - it is detected correctly now
And the color option has gone
All good now - thanks!

**Reply by Diaoul** [reply]
@taugusti do you confirm the network instabilities are gone following the upgrade? :pray: 

I just bought a bunch of those Eglo lights so I never saw what was wrong first hand, I'd rather give them back if they still cause issues :grimacing:

**Reply by taugusti** [reply]
Hi @Diaoul 
The lamps works much better with the upgrade. No network errors and issues like that.
But they still suffer from not always responding when they have not been operated for extended periods of time...
Honestly I would not buy them again.
best of luck
/th

**Reply by d3relict** [reply]
i have an awox 33955 in my network that was firmware upgraded to 2.7.x. before the upgrade, the unit sent frequent updates, like every 10-30 seconds or so. after the upgrade, the updates are sent every few hours, which seems reasonable, and also seems more responsive (but still a bit slow every now and then). strange thing is, that zigbee2mqtt still reports the device with firmware version 2.3.12_250, but the awox app shows another (and no further option to upgrade).

**Reply by remidebette** [reply]
If someone upgrades and sees this version mismatch too, just interview the device and the version will be corrected

**Comment by 87insane** [comment]
Hey guys,
I also have an EGLO LED (https://www.amazon.de/dp/B09FSY66SH?psc=1&ref=ppx_yo2ov_dt_b_product_details). I noticed that a set off does not switch off the LED but sets the brightness to 1. The LED can be switched on/off completely via the app (HomeControl).

Info: My light is an RGBCCT

`{"id":12,"type":"Router","ieeeAddr":"0xa4c1384e1fbf357a","nwkAddr":61468,"manufId":4417,"manufName":"AwoX","powerSource":"Mains (single phase)","modelId":"EGLO_ZM_RGB_TW","epList":[1,3,242],"endpoints":{"1":{"profId":260,"epId":1,"devId":269,"inClusterList":[0,3,4,5,6,8,768,4096,64599,10],"outClusterList":[6],"clusters":{"genBasic":{"attributes":{"swBuildId":"2.5.0_999","zclVersion":3}},"lightingColorCtrl":{"attributes":{"colorCapabilities":25,"colorTempPhysicalMin":153,"colorTempPhysicalMax":370,"colorMode":2,"currentX":9122,"currentY":3559,"colorTemperature":304,"startUpColorTemperature":370}},"genLevelCtrl":{"attributes":{"currentLevel":1}},"genOnOff":{"attributes":{"onOff":1,"startUpOnOff":1}}},"binds":[],"configuredReportings":[],"meta":{"onLevelSupported":false}},"3":{"profId":4751,"epId":3,"devId":269,"inClusterList":[65360,65361,4],"outClusterList":[65360,65361],"clusters":{},"binds":[],"configuredReportings":[],"meta":{}},"242":{"profId":41440,"epId":242,"devId":97,"inClusterList":[],"outClusterList":[33],"clusters":{},"binds":[],"configuredReportings":[],"meta":{}}},"appVersion":2,"stackVersion":1,"hwVersion":60,"swBuildId":"2.5.0_999","zclVersion":3,"interviewCompleted":true,"meta":{"configured":88764544},"lastSeen":1690809250174,"defaultSendRequestWhen":"immediate"}`

**Reply by jeroenvansch** [reply]
Hi, same behaviour here for a eglo downlights that identify as Awox 33957. 
Since updating to 2.7.0 the turn on/off doesnt work correctly. A remote that has been bound to the light does work as expected.
so I think there is something wrong with the converter.

**Reply by Diaoul** [reply]
I have the same issue. How can I help get this solve? I can contribute code if needed but pointers would be appreciated :pray:

**Reply by taugusti** [reply]
I think you should create a ned issue and link to this one. This one is closed so the right people might not see it...
/Th

**Reply by Diaoul** [reply]
:tada: I've managed to fix this issue (it seems) by emptying the `Transition` in the `Setting (specific)` tab.

It was previously at 1 as I wanted transitions on every change. I even tried to set it back to 0 which is supposed to be the default but it did not work either. I imagine `0` and `empty` have different meanings :shrug: 

Edit: I think it's still actually a bit broken when trying to use transitions.

**Reply by 87insane** [reply]
I created a ticket: https://github.com/Koenkk/zigbee2mqtt/issues/19007

**Comment by Ruud-cb** [comment]
Is the firmware update available through Home Assistant already? I still having these issues with firmware version 2.7.3_1387, after a few commands it does not respond anymore.

Device type
Router
Zigbee Model
EGLO_ZM_RGB_TW
Zigbee Manufacturer
AwoX
Description
RGB light
Support status
Supported
IEEE Address
Oxa4c138e16097f363 Network address
Ox2429 / 9257.
Firmware version
2.7.3_1387
Manufacturer
EGLO
Model
900024/12253

**Reply by taugusti** [reply]
I do not think so
I had to install the app, connect through bluetooth and update from there.

I still run the 2.5.0 fw

**Reply by Ruud-cb** [reply]
I updated via the app, now running 2.7.6_1477 yet nothing has changed, still laggy. Directly via the Awox Homecontrol app is super fast (Bluetooth i guess). So I guess it's Zigbee2Mqtt related? How can I debug this. I tried to specifically permit-join from a nearby router (switch_kitchen_right...), and as you can see my network is not really big yet. Any ideas?
<img width="1440" height="3120" alt="Screenshot_20250716-190337" src="https://github.com/user-attachments/assets/e403312c-918e-481a-84ee-1427ef50fa5b" />

**Reply by jeroenvansch** [reply]
What coordinator hardware and firmware are you using?

**Reply by Ruud-cb** [reply]
@jeroenvansch SMZB-06 PoE, updated to the latest firmware.
I do notice a slight improvement when moving the light closer to the Coordinator, but on the other side I have a normal gu10 spotlight that can do white-balance and brightness close to the normal location which response is very fast. I actually don't understand why the light is not talking via the switch_kitchen_right router, is it better to have a poor network quality to the Coordinator then a excellent network quality to the closest router?
Reason for trying to stick with this one is due the limited options of a 68mm black ceiling spotlight..

**Reply by Ruud-cb** [reply]
After some more digging I've seen these logs:
```
[2025-07-17 20:14:19] info:     z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/light_diningroom_spot1', payload '{"brightness":133,"color_mode":"color_temp","color_temp":370,"effect":null,"linkquality":29,"power_on_behavior":null,"state":"ON"}'
[2025-07-17 20:14:20] info:     z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/light_diningroom_spot1', payload '{"brightness":130,"color_mode":"color_temp","color_temp":370,"effect":null,"linkquality":32,"power_on_behavior":null,"state":"ON"}'
[2025-07-17 20:14:20] info:     z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/light_diningroom_spot1', payload '{"brightness":124,"color_mode":"color_temp","color_temp":370,"effect":null,"linkquality":25,"power_on_behavior":null,"state":"ON"}'
[2025-07-17 20:14:44] error:    z2m: Publish 'set' 'brightness' to 'light_diningroom_spot1' failed: 'Error: ZCL command 0xa4c138e16097f363/1 genLevelCtrl.moveToLevelWithOnOff({"level":114,"transtime":0}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"reservedBits":0,"writeUndiv":false}) failed (Timeout - 29849 - 1 - 90 - 8 - 11 after 10000ms)'
[2025-07-17 20:14:44] info:     z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/light_diningroom_spot1', payload '{"brightness":105,"color_mode":"color_temp","color_temp":370,"effect":null,"linkquality":29,"power_on_behavior":null,"state":"ON"}'
[2025-07-17 20:14:44] info:     z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/light_diningroom_spot1', payload '{"brightness":95,"color_mode":"color_temp","color_temp":370,"effect":null,"linkquality":32,"power_on_behavior":null,"state":"ON"}'
```

I can push more commands all I like but it basically waits for this timeout to occur and then process new commands.
Like I said, it 'hangs', so after this timeout it is able to process more commands, but they all 'built up'.
I guess I either have to reduce the timeout (how?) or return the light and find something else...
To be more specific on the firmware: 2.9.4 I'm running on the SLZB-06. and 2.5.1 Zigbee2mqtt, running from Alpine LXC Proxmox server.
Any help or suggestions are welcome.

**Comment by blitzu** [comment]
I have an update of firmware to 3.0.0, it looks that it is ok now. Maybe you can check

**Reply by pullasuti** [reply]
Usually I'm skeptical about these kinds of updates. I updated to the latest version (2.xx back then) last fall and my network was still breaking and lagging so bad, I removed the single Eglo light ([AwoX 12226](https://www.zigbee2mqtt.io/devices/12226.html#awox-12226)) I had. 

This week thought about trying again. FW 3.0.1 was available, and the Awox HomeControl app specifically listed compatibility with 3rd party coordinators (don't remember the exact wording) as an improvement. 

The bulb has been connected for two days now without any issues with my mesh, and it responds completely without lag. 

It seems (at least in my case) that the 3.0 firmware is good enough to actually consider buying Awox/Eglo products :thinking: 

Maybe they are motivated to fix Zigbee, as it seems BT support is being deprecated in their app (Android?).
